Dockerfile

```diff
@@ -70,142 +78,121 @@ RUN sudo apt install -y --no-install-recommends \
-
-
-## ROS Installation
-WORKDIR /home/$USERNAME
-# RUN sudo apt update --fix-missing && sudo apt install ros-melodic-desktop-full  -y --no-install-recommends
-RUN sudo apt-get update && sudo apt-get install ros-melodic-desktop-full  -y --no-install-recommends
-# RUN sudo apt-get update && sudo apt-get install ros-melodic-desktop-full ros-melodic-perception ros-melodic-simulators ros-melodic-urdf-sim-tutorial ros-melodic-perception-pcl ros-melodic-gazebo-ros-pkgs -y --no-install-recommends
-# dependencies from Wenshan Doc
-RUN sudo apt install ros-melodic-octomap ros-melodic-octomap-mapping ros-melodic-octomap-msgs ros-melodic-octomap-ros ros-melodic-octomap-rviz-plugins ros-melodic-octomap-server ros-melodic-dynamic-edt-3d -y --no-install-recommends
-# # dependencies for building packages (TODO: necessary?)
-RUN sudo apt install ros-melodic-catkin ros-melodic-teleop-twist-keyboard python-pip python-wstool python-catkin-tools -y --no-install-recommends
-# RUN sudo apt install python-rosdep python-rosinstall python-rosinstall-generator python-wstool build-essential -y --no-install recommends
-
-
-# ## AirSim Installation Dependencies (TODO: necessary?)
-# # # RUN sudo apt-get install python3 python3-pip python3-dev sudo libglu1-mesa-dev xdg-user-dirs pulseaudio -y --no-install-recommends
-# # RUN sudo apt install python3 python3-pip python3-dev sudo libglu1-mesa-dev xdg-user-dirs -y --no-install-recommends
-# # RUN sudo apt install build-essential cmake cppcheck gdb git vim wget tmux less htop python python-pip python-tk -y --no-install-recommends
-
-# Cleanup
-RUN sudo apt clean autoremove
-
-# # ## Python (TODO: necessary?)
-# # RUN pip3 install setuptools wheel --no-cache-dir
-# # RUN pip3 install 'ue4cli>=0.0.41' 'conan-ue4cli>=0.0.21' ue4-ci-helpers --no-cache-dir
-# # RUN pip3 install msgpack-rpc-python numpy airsim --no-cache-dir
-# # RUN pip install colored-traceback catkin_tools msgpack-rpc-python torch future --no-cache-dir
-
-RUN sudo chown -R $USERNAME /home/$USERNAME
-## Unreal Setup (https://bitbucket.org/castacks/cluster/wiki/airsim_ros)
-# ( https://microsoft.github.io/AirSim/build_linux/)
+RUN git clone -b 4.25 https://$GITHUB_USER:$GITHUB_PWD@github.com/EpicGames/UnrealEngine.git
 WORKDIR /$FOLDER_NAME/UnrealEngine
-# RUN sudo chown -R $USERNAME /$FOLDER_NAME
 RUN ./Setup.sh
 RUN ./GenerateProjectFiles.sh
 USER $USERNAME
 RUN make
-WORKDIR /$FOLDER_NAME
 
-## AirSim Setup (https://microsoft.github.io/AirSim/build_linux/)
-WORKDIR AirSim
+
+# =======================================================================
+#  AirSim v1.4.0-linux (https://microsoft.github.io/AirSim/build_linux/)
+# =======================================================================
+WORKDIR /$FOLDER_NAME
+RUN git clone --depth 1 -b v1.4.0-linux https://github.com/Microsoft/AirSim.git
+WORKDIR /$FOLDER_NAME/AirSim
 RUN ./setup.sh
 RUN ./build.sh
 # use ./build.sh --debug to build in debug mode
+# TODO: Copy Plugins folder to host if possible
```

Comment:
How come need to copy to host?

Reply:
Not really needed, just thought to "auto" copy to host when we build this docker image, so we can just add this Plugin folder to our own UE envs. Will remove this TODO.

Reply:
Removed.